src/AutoMapper/TypeMap.cs

```diff
@@ -180,11 +180,18 @@ public string[] GetUnmappedPropertyNames()
                     .Where(smc => smc.IsIgnored())
                     .Select(pm => pm.SourceMember.Name).ToList();
 
+                var redirectedPathMembers = PathMaps
```

Comment:
Yes, that's the gist of it. But it's kind of ad-hoc. We have this common interface IMemberMap, so I think implementing smth through that interface would be better. So the end result would be smth like `PropertyMaps.Concat(PathMaps).Where`. Doable I think.

Reply:
Yeah, the `TypeMap.MemberMaps` property does this very well. However, I don't think that there's a smart solution (or I didn't understand the codebase enough) using the interface.

Mainly because the information about the source types is saved in totally different places. For `PropertyMap`s, I would use `PropertyMap.SourceMember` while for the `PathMap`, I evaluate an expression. Of course, the `TypeMap` property is common to both types, however I personally don't like the idea of using Reflection again, when it was already done at a previous place (in the `PropertyMap._memberChain` list) for one of the types.

Reply:
You can add SourceMember to IMemberMap and implement it in a similar way to PropertyMap.